modules/programs/wlogout.nix

```diff
@@ -0,0 +1,160 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    all filterAttrs hasAttr isStorePath literalExpression optionalAttrs types;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (builtins) toJSON concatStringsSep;
+
+  cfg = config.programs.wlogout;
+
+  jsonFormat = pkgs.formats.json { };
+
+  mkMargin = name:
+    mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      example = 10;
+      description = "Margin value without unit.";
+    };
+
+  wlogoutLayoutConfig = with types;
+    submodule {
+      freeformType = jsonFormat.type;
+
+      options = {
+        label = mkOption {
+          type = str;
+          default = "";
+          example = "shutdown";
+          description = "CSS label of button.";
+        };
+
+        action = mkOption {
+          type = str;
```

Comment:
I would allow either a string or a path to pass scripts directly e.g. `action = pkgs.writeShellScript "foo" ''bar''`.